app/scripts/migrations/122.ts

```diff
@@ -0,0 +1,59 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 122;
+
+/**
+ * This migration sets preference redesignedConfirmationsEnabled to true
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+// TODO: Replace `any` with specific type
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function transformState(state: Record<string, any>) {
+  if (!hasProperty(state, 'PreferencesController')) {
+    return state;
+  }
+
+  if (!isObject(state.PreferencesController)) {
+    const controllerType = typeof state.PreferencesController;
+    global.sentry?.captureException?.(
+      new Error(`state.PreferencesController is type: ${controllerType}`),
+    );
+    state.PreferencesController = {
+      preferences: {
+        redesignedConfirmationsEnabled: false,
+      },
+    };
+  }
+
+  if (
+    state.PreferencesController.preferences.redesignedConfirmationsEnabled ===
+      false ||
+    state.PreferencesController.preferences.redesignedConfirmationsEnabled ===
+      undefined
+  ) {
+    state.PreferencesController.preferences.redesignedConfirmationsEnabled =
+      true;
+  }
+
+  return state;
```

Comment:
Are the returns redundant since we don't use the return value?

Reply:
Great callout @matthewwalsh0! It seems so 🤔  I was following existing patterns to be safe. I dug into this a bit more. There was a time when transformState would create a new state object and return the new state object to be used (see migrations 007 or 008). That said, this may have been misleading since I found instances that did not do a deep copy and shared prop objects. 

Transforming the state directly, as implied by the function name, and removing the return statement sounds good to me. A new age in our migrations applied here 921922abfcf102b435d23b38c6f779bbe49a616e